gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientImpl.java

```diff
@@ -494,6 +500,98 @@ public void close() {
     }
   }
 
+  @Override
+  public List<GoogleCloudStorageItemInfo> updateItems(List<UpdatableItemInfo> itemInfoList)
+      throws IOException {
+    logger.atFiner().log("updateItems(%s)", itemInfoList);
+
+    if (itemInfoList.isEmpty()) {
+      return new ArrayList<>();
+    }
+
+    Map<StorageResourceId, GoogleCloudStorageItemInfo> resultItemInfos = new ConcurrentHashMap<>();
+    Set<IOException> innerExceptions = newConcurrentHashSet();
+    BatchExecutor executor = new BatchExecutor(storageOptions.getBatchThreads());
+
+    for (UpdatableItemInfo itemInfo : itemInfoList) {
```

Comment:
let us do this before creating executors etc.

Reply:
Done